aspnetcore/fundamentals/middleware/extensibility.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Middleware activation with nonconforming containers in ASP.NET Core
+author: guardrex
+description: Learn how to activate middleware with nonconforming containers in ASP.NET Core.
+ms.author: riande
+manager: wpickett
+ms.custom: mvc
+ms.date: 01/17/2018
+ms.topic: article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: fundamentals/middleware/extensibility
+---
+# Middleware activation with nonconforming containers in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+[IMiddlewareFactory](/dotnet/api/microsoft.aspnetcore.http.imiddlewarefactory)/[IMiddleware](/dotnet/api/microsoft.aspnetcore.http.imiddleware) is an extensibility point for [middleware](xref:fundamentals/middleware/index) activation with nonconforming containers.
+
+`UseMiddleware` extension methods check if a middleware's registered type implements `IMiddleware`. If it does, the `IMiddlewareFactory` instance registered in the nonconforming container is used to resolve the `IMiddleware` implementation instead of using the convention-based middleware activation logic. The factory is registered as a scoped service with the app's noncomforming service container.
```

Comment:
Remove the _nonconforming container_ statement. Just say container.

Reply:
Are we removing "nonconforming" everywhere ... from the title as well?